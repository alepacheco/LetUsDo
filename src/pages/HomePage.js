import React from 'react';
// import { Link } from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import Form from 'react-bootstrap/Form';

const HomePage = () => {
  return (
    <div className="home-page center-v-h">
      <h1 className="main-text">We Do</h1>
      <p className="sub-text">Lorem isupm </p>
      <Card>
        <Form className="cccc">
          <Form.Group controlId="exampleForm.ControlTextarea1">
            <Form.Label>What task do you need done?</Form.Label>
            <Form.Control as="textarea" rows="3" />
          </Form.Group>
        </Form>
        <Button className="get-it-done-button">Get it done</Button>
      </Card>
    </div>
  );
};

export default HomePage;
