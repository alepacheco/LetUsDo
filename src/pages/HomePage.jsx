import React, { Suspense, lazy } from 'react';

import { LandingHeader } from '../components/LandingHeader';
import { FeaturesList } from '../components/FeaturesList';
import { Footer } from '../components/Footer';

const TaskModal = lazy(() => import('../components/TaskModal'));
const FullFeatureList = lazy(() => import('../components/FullFeatureList'));
const Loading = () => <div>Loading...</div>;

const HomePage = () => (
  <div>
    <LandingHeader />
    <FeaturesList />
    <Suspense fallback={Loading}>
      <TaskModal />
    </Suspense>
    <Suspense fallback={Loading}>
      <FullFeatureList />
    </Suspense>
    <Footer />
  </div>
);

export default HomePage;
