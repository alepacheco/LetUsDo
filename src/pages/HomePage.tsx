import React, { Suspense, lazy } from 'react';
// @ts-ignore
import { LandingHeader } from '../components/LandingHeader';
// @ts-ignore
import { FeaturesList } from '../components/FeaturesList';
// @ts-ignore
import { Footer } from '../components/Footer';
// @ts-ignore
import CheckoutPopUp from '../components/CheckoutPopUp/index';
// @ts-ignore
const TaskBox = lazy(() => import('../components/TaskBox'));
// @ts-ignore
const FullFeatureList = lazy(() => import('../components/FullFeatureList'));
const Loading = <div>Loading...</div>;

const HomePage = () => (
  <div>
    <LandingHeader />
    <FeaturesList />
    <Suspense fallback={Loading}>
      <TaskBox />
      <CheckoutPopUp />
    </Suspense>
    <Suspense fallback={Loading}>
      <FullFeatureList />
    </Suspense>
    <Footer />
  </div>
);

export default HomePage;
