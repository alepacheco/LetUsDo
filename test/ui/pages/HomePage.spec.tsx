/* eslint-disable no-undef, import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from 'src/../pages/index';

describe('<HomePage />', () => {
  it('renders', () => {
    const wrapper = shallow(<HomePage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
