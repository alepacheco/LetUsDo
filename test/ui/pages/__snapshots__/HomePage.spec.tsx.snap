// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders 1`] = `
<div>
  <LandingHeader />
  <Component />
  <Suspense
    fallback={
      <div>
        Loading...
      </div>
    }
  >
    <lazy />
    <lazy />
  </Suspense>
  <Suspense
    fallback={
      <div>
        Loading...
      </div>
    }
  >
    <lazy />
  </Suspense>
  <Footer />
</div>
`;
